src/pcre2_compile.c

```diff
@@ -4745,6 +4949,7 @@ while (ptr < ptrend)
           &i, &errorcode)) goto FAILED;
       if (i < 0)  /* NB (?0) is permitted */
         {
+        // XXX Can we change this block to be PCRE2_ASSERT(i >= 0)? How can read_number return a negative value?
```

Comment:
TODO?

Reply:
Ah yes, an "XXX" comment is a note-to-myself which I will fix before this PR is finished.

But it's also a question to the reviewers. Do we actually need this `i < 0` block, or is impossible now?

Reply:
It does not seems so. Perhaps something was changed.

Reply:
Funny thing is that we usually don't "review" each others code that much.

Reply:
I'm very grateful Carlo (and Zoltan) for the reviews you've been giving me. I'm learning a lot about the PCRE2 history, coding standards, and so on, which is very helpful as a new joiner.

For a code owner making changes to their own code, detailed review isn't always essential. For me though, changing core historic code, it's very valuable.

Reply:
I have removed this block, and replaced it with an assertion. I conclude that it's not reachable.